sdk/core/System.ClientModel/src/Options/PipelineMessageClassifier.cs

```diff
@@ -13,16 +13,27 @@ public static PipelineMessageClassifier Create(ReadOnlySpan<ushort> successStatu
     protected internal PipelineMessageClassifier() { }
 
     /// <summary>
-    /// Specifies if the response contained in the <paramref name="message"/> is not successful.
+    /// Determines whether the Response property of <paramref name="message"/>
+    /// is considered an error response by pipeline policies and the client.
     /// </summary>
-    public virtual bool IsErrorResponse(PipelineMessage message)
+    /// <param name="message">The message to classify.</param>
+    /// <param name="isError">True if the classifer considers the message's
+    /// response to be an error; false otherwise.</param>
+    /// <returns>True if the classifier provided a value for <paramref name="isError"/>,
+    /// false if it did not classify the message response. A classifier can return
+    /// false if it wants to be composable with other classifiers and pass the
+    /// decision on to another classifier instead of doing the classification
+    /// itself.</returns>
+    public virtual bool TryClassifyResponse(PipelineMessage message, out bool isError)
```

Comment:
@JoshLove-msft -- I think the challenge here will be making it so that any derived type's implementation of IsErrorResponse is respected, e.g. if someone overrides IsErrorResponse on the Azure.Core classifier, the base ClientModel classifier uses that implementation rather than providing a second one.  I think we will need to provide an API in addition to this TryClassify one that lets the derived type say "I want to classify this status code" -- but I'm worried that will start to make the classifier API overly complex.

Reply:
Hmm how would that work if this type no longer defines IsErrorResponse?

Reply:
Maybe we could just hide IsErrorResponse from ResponseClassifier and say TryClassifyResponse is the method to implement going forward.